File: .semgrep/no-sensitive-values-in-logs.py
import logging

logger = logging.getLogger(__name__)
api_key = "x"
token = "x"
user_email = "x"
count = 3
url = "https://example.test/"


def log(message: str) -> None:
    pass


class Client:
    def log(self, message: str) -> None:
        pass

    def run(self) -> None:
        # ruleid: no-sensitive-values-in-logs-python
        self.log(f"retrying with {token}")
        # ok: no-sensitive-values-in-logs-python
        self.log(f"network error on {url}; retry {count}")


# ruleid: no-sensitive-values-in-logs-python
print(api_key)
# ruleid: no-sensitive-values-in-logs-python
print(f"key is {api_key}")
# ruleid: no-sensitive-values-in-logs-python
log(f"user {user_email} signed in")
# ruleid: no-sensitive-values-in-logs-python
logger.info("auth %s", token)
# ok: no-sensitive-values-in-logs-python
log(f"shows: {count} fetched this run")
# ok: no-sensitive-values-in-logs-python
print(f"ok: {url} still grants reuse")
# ok: no-sensitive-values-in-logs-python
logger.info("fetched %d", count)
tokens: list[str] = []
# ok: no-sensitive-values-in-logs-python
tokens.append(token)

File: .semgrep/no-sensitive-values-in-logs.swift
import Foundation

func demo(authToken: String, count: Int, userEmail: String) {
    // ruleid: no-sensitive-values-in-logs-swift
    print(authToken)
    // ruleid: no-sensitive-values-in-logs-swift
    NSLog("%@", userEmail)
    // ok: no-sensitive-values-in-logs-swift
    print(count)
    var offenders: [String] = []
    // ok: no-sensitive-values-in-logs-swift
    offenders.append("\(authToken)")
    let logger = Logger()
    // ruleid: no-sensitive-values-in-logs-swift
    logger.error(authToken)
}

struct Logger {
    func error(_ message: String) {}
}
